src/backend/distributed/planner/function_call_delegation.c

```diff
@@ -217,10 +269,29 @@ TryToDelegateFunctionCall(DistributedPlanningContext *planContext)
 	{
 		/*
 		 * We are planning a call to a distributed function within a Citus backend,
-		 * that means that this is the delegated call.
+		 * that means that this is the delegated call. If the function is forcefully
+		 * delegated, capture the distribution argument.
 		 */
-		InDelegatedFunctionCall = true;
-		return NULL;
+		CheckDelegatedFunctionExecution(procedure, funcExpr);
+
+		/*
+		 * InSafeDelegatedFunctionCall flag grants the levy to do remote tasks from
+		 * a delegated function.
+		 */
+		if (procedure->forceDelegation && IsMultiStatementTransaction())
+		{
+			/*
+			 * It's not safe when the function is delegated with an explicit
+			 * transaction on the client session.
+			 */
+			InSafeDelegatedFunctionCall = false;
```

Comment:
Is there any side-effect to setting this to false here? e.g. if we have forced function call delegation within a delegated function call?

Reply:
We are doing only at the top level now.